js_modules/dagit/packages/ui/src/components/MiddleTruncate.tsx

```diff
@@ -0,0 +1,126 @@
+import useResizeObserver from '@react-hook/resize-observer';
```

Comment:
Should we use https://github.com/dagster-io/dagster/blob/master/js_modules/dagit/packages/core/src/gantt/useViewport.tsx instead?
We can add a flag to only enable the resize observer and disable monitoring scroll position/offset.

Reply:
Oh you can actually just not spread `onScroll` https://github.com/dagster-io/dagster/blob/master/js_modules/dagit/packages/core/src/gantt/useViewport.tsx#L159

Reply:
The lib I used here uses a single ResizeObserver for all observed elements, instead of a new ResizeObserver per element. Admittedly I don't know if the perf benefits are the same today as when the lib hook was written.

Perhaps `useViewport` could be updated to do the same, or to use the lib hook directly...